xscen/io.py

```diff
@@ -518,6 +488,29 @@ def coerce_attrs(attrs):
             raise
 
 
+def _rechunk_for_saving(ds, rechunk):
+    """Do chunking before saving to .zarr or .nc, generalized as Y/X for different axes lat/lon, rlat/rlon"""
+    for rechunk_var in ds.data_vars:
+        # Support for chunks varying per variable
+        if rechunk_var in rechunk:
+            rechunk_dims = rechunk[rechunk_var].copy()
+        else:
+            rechunk_dims = rechunk.copy()
+
+        # get actual axes labels
+        if {"X", "Y"}.issubset(rechunk_dims):
+            rechunk_dims[ds.cf.axes["X"][0]] = rechunk_dims.pop("X")
+            rechunk_dims[ds.cf.axes["Y"][0]] = rechunk_dims.pop("Y")
```

Comment:
```suggestion
        if "X"  in rechunk_dims and "X" not in ds.dims:
            rechunk_dims[ds.cf.axes["X"][0]] = rechunk_dims.pop("X")
        if "Y"  in rechunk_dims and "Y" not in ds.dims:
            rechunk_dims[ds.cf.axes["Y"][0]] = rechunk_dims.pop("Y")
```
To allow chunking along only one of the dimensions and to do your trick only when there's not a real "X" dimension.

Reply:
Good catch! I like it.